test/parallel/test-timers-enroll-second-time.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const timers = require('timers');
+
+let didCall = false;
+const enrollObj = {
+  _onTimeout: common.mustCall(() => assert(didCall)),
+};
+
+timers.enroll(enrollObj, 1);
+timers.enroll(enrollObj, 10);
```

Comment:
More reliable: put the `setTimeout` before this and then just re-enroll with `1`. The list order should guarantee the call order is as we expect without depending on libuv behavior.